### PR TITLE
Handle api request exceptions gracefully

### PR DESCRIPTION
**Pull request by kxtran**

### CODE REVIEW

File: log10/load.py [hunk 1]
```diff
@@ -283,10 +310,15 @@ def wrapper(*args, **kwargs):
             ]
 
             start_time = time.perf_counter()
+            logger.debug(f"TIMED BLOCK - LLM call start time: {start_time}")
+
```
Comment (wenzhe-log10) [inline log10/load.py:314]: nit: I don't think we need these lines.

File: log10/load.py [hunk 2]
```diff
@@ -310,8 +342,8 @@ def wrapper(*args, **kwargs):
                 "session_id": sessionID,
                 "tags": global_tags,
             }
-            res = post_request(completion_url + "/" + completionID, log_row)
-            raise e
+            if not got_log10_exception:
+                res = post_request(completion_url + "/" + completionID, log_row)
```
Comment (wenzhe-log10) [inline log10/load.py:346]:
```suggestion
                res = post_request(completion_url + "/" + completionID, log_row)
                if res is None or res.status_code != 200:
                    logger.error(f"LOG10: failed to insert in log10: {log_row}.")
                    got_log10_exception = True
```

Comment (wenzhe-log10) [inline log10/load.py:346]:
```suggestion
                res = post_request(completion_url + "/" + completionID, log_row)
                if res is None or res.status_code != 200:
                    logger.error(f"LOG10: failed to insert in log10: {log_row}.")
                    got_log10_exception = True
```
Here may fail and we should log the error.

File: log10/load.py [hunk 3]
```diff
@@ -353,7 +386,7 @@ def wrapper(*args, **kwargs):
                     "tags": global_tags,
                 }
 
-                if target_service == "log10":
+                if target_service == "log10" and not got_log10_exception:
                     res = post_request(completion_url + "/" + completionID, log_row)
                     if res.status_code != 200:
```
Comment (wenzhe-log10) [inline log10/load.py:391]:
```suggestion
                    if res is None or res.status_code != 200:
```

File: log10/llm.py [hunk 4]
```diff
@@ -226,17 +234,18 @@ def log_end(self, completion_id: str, response: dict, duration: int):
...
-                "stacktrace": json.dumps(stacktrace),
-            },
-        )
+        if not self.got_log10_exception:
+            self.api_request(
+                f"/api/completions/{completion_id}",
+                "POST",
+                {
+                    "organization_id": self.log10_config.org_id,
+                    "response": json.dumps(response),
```
Comment (nqn) [inline log10/llm.py:243]: We are actually missing to resend the request here

**Pull request closed without merging**